src/views/Pagination/PaginationContainer.jsx

```diff
@@ -0,0 +1,76 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { getPages } from '../../redux/actions/postActions';
+
+class PaginationContainer extends Component {
+  componentDidMount() {
+    const { getPages } = this.props;
+    getPages();
+  }
+  render() {
+    const {
+      handleLink, handleClick, pageNumbers, currentPage,
+    } = this.props;
+    const {post:{pages}}=this.props;
+    return (
+      <div className="pagination">
+        <nav aria-label="Page navigation">
+          <ul className="pagination">
+            {pages.previous ? (
+              <li className="page-item">
+                <span
+                  className="page-link"
+                  id={pages.previous}
+                  onClick={handleLink}
+                  aria-label="Previous"
+                  role="presentation"
+                >
+                  <span aria-hidden="true">&laquo;</span>
+                  <span className="sr-only">Previous</span>
+                </span>
+              </li>
+            ) : null}
+
+            {pageNumbers.map((num, i) => (
+              <li key={i} className="page-item">
```

Comment:
Hello @Amoswachira ,
It is not advisable to use array indices as keys in lists. You can use the index in combination with other values like we had done before when fetching all articles.

Reply:
@Ogutu-Brian let me fix that

Reply:
@Ogutu-Brian sloved